### PR TITLE
fix: Enable Salesforce & CRMs for teams & orgs [CAL-2116]

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR adds state that contains a teamId in order to install CRM apps to teams.

Fixes # (issue)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Create a team
- Install Salesforce
- Create a booking
- The new attendee should be created in Salesforce

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/features/bookings/lib/handleNewBooking.ts
Comment (joeauyeung) [inline packages/features/bookings/lib/handleNewBooking.ts:656]: We need to pass `teamCredentials` to the event manager in order to trigger any CRM apps

File: packages/app-store/salesforce/lib/CalendarService.ts
Comment (joeauyeung) [inline packages/app-store/salesforce/lib/CalendarService.ts:82]: Requires a last name but need to fallback in case the attendee only supplies a first name

**Review by leog — DISMISSED** [review]
Tested locally and it works, great job

File: packages/features/bookings/lib/handleNewBooking.ts
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking.ts:873]: Maybe move it to a separate function and call it here. The handler function is too big already

File: packages/features/bookings/lib/handleNewBooking.ts
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking.ts:1166]: Better to create a variable(maybe possibleCredentials) and use that everywhere.

File: packages/features/bookings/lib/handleNewBooking.ts
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking.ts:1890]: The logic downstream is going to pick the latest added credential first and thus there is no visibility and direct control on which credential would be picked. 

So, from a user perspective if there is Zoom app installed on Team, Org, and organizer level and the eventType for the organizer is booked, the meeting can go to any of the Team, Org, organizer connected accounts and it would keep on varying depending on which was installed most recently.


I think we should provide a way to be very clearly select which credential to use for an App for an EventType

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Have added some comments. Waiting on @Jaibles's input here and can be merged after.

**Review by hariombalhara — APPROVED** [review]
Merging this PR as it's mostly working as expected. I have some confusions that I have pinged @joeauyeung about but I don't want to block the merge as we have a huge timezone gap and that would slow down things.

**Pull request merged**
